projects/plugins/crm/src/automation/commons/conditions/tags/class-object-tag.php

```diff
@@ -0,0 +1,261 @@
+use Automattic\Jetpack\CRM\Automation\Base_Condition;
+use Automattic\Jetpack\CRM\Automation\Data_Types\Data_Type;
+use Automattic\Jetpack\CRM\Automation\Data_Types\Tag_Data;
+
+/**
+ * Object Tag condition class.
+ *
+ * @since $$next-version$$
+ */
+class Object_Tag extends Base_Condition {
+
+	/**
+	 * Object Tag constructor.
+	 *
+	 * @since $$next-version$$
+	 *
+	 * @param array $step_data The step data.
+	 */
+	public function __construct( array $step_data ) {
+		parent::__construct( $step_data );
+
+		$this->valid_operators = array(
+			'tag_added'   => __( 'Tag was added', 'zero-bs-crm' ),
+			'tag_removed' => __( 'Tag was removed', 'zero-bs-crm' ),
+			'has_tag'     => __( 'Has tag', 'zero-bs-crm' ),
+			'not_has_tag' => __( 'Does not have tag', 'zero-bs-crm' ),
+		);
+
+		$this->set_attribute_definitions(
+			array(
+				new Attribute_Definition( 'operator', __( 'Operator', 'zero-bs-crm' ), __( 'Determines how the field is compared to the specified value.', 'zero-bs-crm' ), Attribute_Definition::SELECT, $this->valid_operators ),
+				new Attribute_Definition( 'tag', __( 'Tag', 'zero-bs-crm' ), __( 'Object Tag to compare with.', 'zero-bs-crm' ), Attribute_Definition::TEXT ),
+			)
+		);
+	}
+
+	/**
+	 * Check for valid parameters.
+	 *
+	 * @since $$next-version$$
+	 *
+	 * @param string $operator The operator.
+	 * @param mixed  $data The data to validate.
+	 * @param mixed  $previous_data The previous data to validate.
+	 * @return bool True if parameters are valid, false otherwise.
+	 */
+	private function check_for_valid_parameters( $operator, $data, $previous_data ) {
+
+		switch ( $operator ) {
+			case 'tag_added':
+			case 'tag_removed':
+				if ( $previous_data === null || ! $this->is_valid_tag_data( $data ) || ! $this->is_valid_tag_data( $previous_data ) ) {
+					$this->logger->log( 'Invalid tag data: parameters not valid' );
+
+					return false;
+				}
+				break;
+			case 'has_tag':
+			case 'not_has_tag':
+				if ( ! $this->is_valid_tag_data( $data ) ) {
+					$this->logger->log( 'Invalid tag data: parameters not valid' );
+					$this->condition_met = false;
+
+					return false;
+				}
+				break;
+			default:
+				$this->logger->log( 'Unknown operator: ' . $operator );
+				return false;
+		}
+
+		return true;
+	}
+
+	/**
+	 * Checks if a given tag name exists in the tags array.
+	 *
+	 * @param array  $data     The array containing tag data.
+	 * @param string $tag_name The name of the tag to check for.
+	 *
+	 * @return bool True if the tag name exists, false otherwise.
+	 */
+	private function has_tag_by_name( $data, string $tag_name ) {
+		if ( ! is_array( $data ) ) {
+			$this->logger->log( 'The given tag data is not in the correct (array) format, so is not valid.' );
+			return false;
+		}
+		foreach ( $data as $item ) {
+			if ( is_array( $item ) && $item['name'] === $tag_name ) {
+				return true;
+			}
+		}
+		$this->logger->log( 'Tag does not exist in the given array of tags: ' );
```

Comment:
I don't think we have to log that information. It's not relevant for any general debugging. If a tag doesn't exist, is normal.